render/respond.go

```diff
@@ -0,0 +1,111 @@
+package render
+
+import (
+	"bytes"
+	"encoding/json"
+	"encoding/xml"
+	"net/http"
+	"reflect"
+
+	"golang.org/x/net/context"
+)
+
+var Respond = DefaultRespond
+
+func DefaultRespond(ctx context.Context, w http.ResponseWriter, v interface{}) {
+	// Present the object.
+	if presenter, ok := ctx.Value(presenterCtxKey).(Presenter); ok {
+		v = presenter.Present(ctx, v)
+	} else {
+		v = DefaultPresenter.Present(ctx, v)
+	}
+
+	// Format data based on Content-Type.
+	switch contentType, _ := ctx.Value(contentTypeCtxKey).(ContentType); contentType {
+	case ContentTypeJSON:
+		JSON(ctx, w, v)
+	case ContentTypeXML:
+		XML(ctx, w, v)
+	default:
+		JSON(ctx, w, v)
+	}
```

Comment:
It's tricky -- for example if someone passes `map[string]string{...}` or another complex object into the responder, how do you want to render it into HTML or TXT? It's not really possible, unless you accept only `string` type, which I did:

```
render.HTML(string)
render.String(string) // actually I'd like render.PlainText(string) better
```

HTML and PlainText is not really something that could accept `interface{}`.